### ISSUE TITLE
loadingSpinner is displayed when isInfiniteLoading={ false }

### ISSUE THREAD

**Issue by paulpooch** [seatgeek/react-infinite#232]
What is with all the generateComputedProps?  
Why are passed in props copied into component state?  

This is creating situations where state.isInfiniteLoading is true even though I passed in isInfiniteLoading={ false } as a prop - and so the loadingSpinner is visible when nothing is loading.

This would be fine is the below were fixed:

```
  var loadingSpinner = this.computedProps.infiniteLoadBeginEdgeOffset === undefined
      ? null
      : <div ref={(c) => { this.loadingSpinner = c; }}>
        {this.state.isInfiniteLoading ? this.computedProps.loadingSpinnerDelegate : null}
      </div>; // CAN WE USE this.props.isInfiniteLoading ABOVE INSTEAD?
```
Here is an image where the two fields are out of sync.

![image](https://user-images.githubusercontent.com/539514/28890364-d50a2e5c-7794-11e7-9e3a-d9b796823715.png)

**Comment by garetht** [comment]
The behavior of `isInfiniteLoading` was part of the original design which automatically [turned on](https://github.com/seatgeek/react-infinite#function-oninfiniteload) the infinite loading spinner instead of waiting for the user to pass the appropriate prop into `Infinite` to turn the spinner on. It would probably have been better and more intuitive to have made the user turn it on manually, but changing that part of the design will have to wait so it does not break current components.

**Comment by paulpooch** [comment]
OK thanks for the explanation.

I am able to workaround this via CSS hack (adding an isLoading classs to parent which makes spinner display: none).

But certainly let me know if you ever want a PR for this.

**Comment by NarasimhaReddyY** [comment]
I encountered the same problem today. It's better to let the user to pass isLoading prop instead of auto setting to true. Thanks

**Comment by torstenrudolf** [comment]
yes, I just ran into this myself, and this is certainly not what you want. it should only use the prop specified. it should only use the inner state, if the prop is undefined.

**Comment by kgregory** [comment]
See [this comment](https://github.com/seatgeek/react-infinite/issues/162#issuecomment-241699280) from @dobryanskyy for an easy workaround:

>I had very similar issue with spinner not getting hidden. I have dealt with that problem by setting `loadingSpinnerDelegate={isLoadingInProgress && <Spinner />}`
